### PR TITLE
Support Loading ONNX model using GPU

### PR DESCRIPTION
**Pull request by tarekgh**

### CODE REVIEW

**Comment by tarekgh** [comment]
@michaelgsharp could you please have a look?

File: src/Microsoft.ML.OnnxTransformer/OnnxTransform.cs [hunk 1]
```diff
@@ -215,6 +215,13 @@ private static OnnxTransformer Create(IHostEnvironment env, ModelLoadContext ctx
                 RecursionLimit = recursionLimit
             };
 
+            IHostEnvironmentInternal localEnvironment = env as IHostEnvironmentInternal;
```
Comment (michaelgsharp) [inline src/Microsoft.ML.OnnxTransformer/OnnxTransform.cs:218]: Do we want this flag to work even if they aren't loading the model? Like if they set these flags should it work when they create the model initially as well?

Comment (tarekgh) [reply src/Microsoft.ML.OnnxTransformer/OnnxTransform.cs:218]: I made all ApplyOnnxModel overloads to use `MLContext.GpuDeviceId` when having the default `gpuDeviceId` parameter.

**Review by michaelgsharp — APPROVED** [review]
:shipit:

**Pull request merged**